### PR TITLE
Creative Chest/Tank fix

### PR DESCRIPTION
**Pull request by brachy84**
Fixes a number format exception.
Makes slot in creatice chest clear on right click

### CODE REVIEW

File: src/main/java/gregtech/api/gui/widgets/PhantomSlotWidget.java [hunk 1]
```diff
@@ -20,14 +22,27 @@
...
+        return this;
+    }
+
     @Override
     public boolean mouseClicked(int mouseX, int mouseY, int button) {
         if (isMouseOverElement(mouseX, mouseY) && gui != null) {
-            gui.getModularUIGui().superMouseClicked(mouseX, mouseY, button);
+            if (button == 1 && clearSlotOnRightClick && !slotReference.getStack().isEmpty()) {
+                slotReference.putStack(ItemStack.EMPTY);
+                writeClientAction(2, buf -> {
```
Comment (bruberu) [inline src/main/java/gregtech/api/gui/widgets/PhantomSlotWidget.java:41]: Should probably be put in GregtechDataCodes.

File: src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCreativeTank.java [hunk 2]
```diff
@@ -71,22 +69,22 @@ protected ModularUI createUI(EntityPlayer entityPlayer) {
                 () -> this.fluidTank.getFluid(), data -> {
             this.fluidTank.setFluid(data);
         })
-                .showTip(false).setBackgroundTexture(GuiTextures.SLOT_DARKENED));
+                .showTip(false));
```
Comment (bruberu) [inline src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCreativeTank.java:72]: The formatting here looks rather off, in my opinion; it should attach to the close parenthesis of the previous line.

**Review by bruberu — COMMENTED** [review]
Just two minor touch-ups; testing seems fine.

File: src/main/java/gregtech/api/gui/widgets/PhantomSlotWidget.java (hunk 1 above)
Comment (brachy84) [reply src/main/java/gregtech/api/gui/widgets/PhantomSlotWidget.java:41]: Not a single widget does it via GregtechDataCodes and the widgets are trash anyways

**Comment by ALongStringOfNumbers** [comment]
Hm, for the quantum Tank, the slot is cleared on left or right click, even though it does not have the `setClearSlotOnRightClick`

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**